dremio_toolkit/env_definition.py

```diff
@@ -17,42 +17,24 @@
 #########################################################################
 
 
+# Data Class to contain definition of Dremio Environment
 class EnvDefinition:
-    containers: list
-    homes: list
-    sources: list
-    spaces: list
-    folders: list
-    vds_list: list
-    reflections: list
-    rules: list
-    queues: list
-    votes: list
-    files: list
-    tags: list
-    wikis: list
-    referenced_users: list
-    referenced_groups: list
-    referenced_roles: list
-    endpoint: str
+    containers = []
+    homes = []
+    sources = []
+    spaces = []
+    folders = []
+    vds_list = []
+    vds_parents = []
+    reflections = []
+    rules = []
+    queues = []
+    votes = []
+    files = []
+    tags = []
+    wikis = []
+    referenced_users = []
+    referenced_groups = []
+    referenced_roles = []
+    endpoint: str = None
```

Comment:
I actually preferred the version with constructor. This provides an interface of how we should manipulate EnvDefinition data class. Usually data classes in scala(case class) or java(lombok.Data) are immutable - They do not change once created. That's why they usually have constructor with all parameters and are immutable afterwards. Here we cannot use @dataclass because it requires python >= 3.8 but we can still use this class as data class. :)